Controllers/UserController.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.IdentityModel.Tokens;
+using project_backend.Models.User;
+using project_backend.Providers;
+using project_backend.Repos;
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+
+namespace project_backend.Controllers
+{
+
+    [ApiController]
+    [Route("[controller]")]
+    public class UserController : ControllerBase
+    {
+        private readonly ILogger<UserController> _logger;
+        private readonly UserProvider _userProvider;
+
+        public UserController(ILogger<UserController> logger, DatabaseContext databaseContext)
+        {
+            _logger = logger;
+            _userProvider = new UserProvider(databaseContext);
+        }
+
+        [HttpPost]
+        [Route("login")]
+        [AllowAnonymous]
+        public string Login([FromBody] UserDTO user)
+        {
+            UserDAO userDAO = new UserDAO
+            {
+                Email = user.Email,
+                Password = user.Password
+            };
+
+            int userId = _userProvider.getUserByCredentials(userDAO);
+            if (userId != -1)
+            {
+                HttpContext.Response.StatusCode = 200;
+                return GenerateToken(userId);
+            }
+            HttpContext.Response.StatusCode = 401;
+            //TODO: find generic a way to treat error messages
+            return "error";
```

Comment:
Poti sa returnezi on obiect Error, cum ai mai jos

Reply:
Unde mai jos?:))) Nu mai bine fac o clasa Response din care sa dea inherit Error,Token si ce mai dam noi inapoi pe response. Si atunci ca return type la functie o sa fie Response ala

Reply:
Scuze.. mai jos era pe ecranu cu files :)) Ai o clasa Error definita in Models, la ea ma refeream.
```public class Error
    {
        public string message { get; set; }
    }```

Reply:
Pai da dar trebuie sa returnez un obiect de tip Token daca totul merge ok sau un obiect de tip Error daca nu merge

Reply:
Ok bun, am gasit o alternativa care ar merge pentru codul asta. Ai putea sa schimbi returnul din string sau whatever in ```IActionResult```, si sa returnezi ```new OkObjectResult(new Token(...))``` daca e succes, sau ```new UnauthorizedObjectResult(new Error(...))```
Ca sa ai si autocomplete pe swagger pentru tipurile astea totusi, trebuie sa specifici cu 2 atribute return type in caz de eroare si in caz normal:
```
[ProducesResponseType(typeof(Token), (int)HttpStatusCode.OK)]
[ProducesErrorResponseType(typeof(Error))]
```